xapian-letor/api/featurevector_internal.cc

```diff
@@ -117,3 +117,31 @@ FeatureVector::Internal::get_nonzero_num(){
     //cout << "nonzero: "<< nonzero <<endl;
     return nonzero;
 }
+
+void
+FeatureVector::Internal::normalise() {
+
+    int num_features = 19;
```

Comment:
Magic number! Should be declared once, or (better?) detected from the size of the map.